hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java

```diff
@@ -381,6 +381,12 @@
 
     void destroy();
 
+    /**
+     * Like {@link #destroy()} but does not touch state on other services
+     * like lock service or event journal service.
+     */
+    void destroyInternals();
```

Comment:
It's getting a bit cofusing - we have `destroy` (destroys the internals and the connected services), `destroyInternals` (destroys only the map internals), `reset` (I'm not sure how this one fits in, it also resets some data but it keeps some other like indexes and connected services like locks and journal) and `clearPartition` (just clears everything, also connected services).
I don't see a simple way of improving this. Maybe a task for a separate PR.

Reply:
Couldn't agree more :100:

Reply:
This is needed for this PR to pass the tests but filed an improvement issue, this problem is a bit tricky. https://github.com/hazelcast/hazelcast/issues/12779